src/expressions/if-expr.md

```diff
@@ -126,30 +104,43 @@ if let E::X(n) | E::Y(n) = v {
 ```
 
 The expression cannot be a [lazy boolean operator expression][_LazyBooleanOperatorExpression_].
-Use of a lazy boolean operator is ambiguous with a planned feature change of the language (the implementation of if-let chains - see [eRFC 2947][_eRFCIfLetChain_]).
-When lazy boolean operator expression is desired, this can be achieved by using parenthesis as below:
 
-<!-- ignore: psuedo code -->
-```rust,ignore
-// Before...
-if let PAT = EXPR && EXPR { .. }
+## Chains of expressions
+
+It is possible to chain multiple expressions into a single statement to avoid nested declarations.
```

Comment:
I think "statement" here is a little confusing.  This also doesn't really present any context, or explain how to chain or how chains work. We generally don't use code blocks as a replacement for explaining how something works, but rather as a helpful illustration and example.

Can you try to add some more detail here? I would expect at least a couple paragraphs explaining what chaining is, what the syntax is, what the semantics are, that you can combine boolean values and `let` pattern bindings, how combining multiple pattern bindings works, etc.

Reply:
Wrote what was suggested in https://github.com/rust-lang/reference/pull/1179#discussion_r850776586